### PR TITLE
Port Surface Processing to `Bikelanes` Dataset

### PR DESCRIPTION
**Pull request by rush42**
This PR ports the surface processing to the `Bikelanes` data set.  We designed this originally for the `roads` data set and now made it globally available as a shared function.

### CODE REVIEW

**Comment by rush42** [comment]
[Here](http://localhost:7800/public.bikelanes.html#16.65/52.529382/13.382424) on Invalidenstr. you can see the Surface and Smoothness being derived from different sources.

File: app/process/legacy_surfaceQuality/surfaceQuality.lua [hunk 1]
```diff
@@ -62,22 +61,13 @@ function osm2pgsql.process_way(object)
     return
   end
 
-  local surface, surface_source = SurfaceDirect(tags.surface)
+  local surface, surface_source = DeriveSurface(tags)
 
   -- Try to find smoothess information in the following order:
   -- 1. `smoothess` tag
   -- 2. `smoothess` extrapolated from surface data
```
Comment (tordans) [inline app/process/legacy_surfaceQuality/surfaceQuality.lua:68]: Move to `DeriveSurface`

**Review by tordans — COMMENTED** [review]
Looks good. Comments added.

File: app/process/legacy_surfaceQuality/surfaceQuality.lua (hunk 1 above)
Comment (rush42) [reply app/process/legacy_surfaceQuality/surfaceQuality.lua:68]: There is already a more detailed version of this comment. I deleted this one from the legacy file

**Pull request merged**